hw/ip/i2c/doc/_index.md

```diff
@@ -142,6 +143,34 @@ Typically every byte transmitted must also receive an ACK signal, and the IP wil
 However, there are some I2C commands which do not require an ACK.
 In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received. 
 
+### Target Address Registers
+
+I2C target device is assigned two 7-bit address and 7-bit mask pairs.
+The target device accepts a transaction if the result of the bitwise AND operation performed on the transaction address sent by the host and a mask matches the assigned address corresponding to the mask.
+Thus, with the masks set to all ones (0x7F), the target device will respond to either of the two assigned unique addresses and no other.
+If the mask and the assigned address both have zeros in a particular bit position, that bit will be a match regardless of the value of that bit received from the host.
+Note that if, in any bit position, the mask has zero and the assigned address has one, no transaction can match and such mask/address pair is effectively diasabled.
+The assigned address and mask pairs are set in registers {{< regref ADDRESS0 >}}, {{< regref MASK0 >}}, {{< regref ADDRESS1 >}}, and {{< regref MASK1 >}}. 
```

Comment:
Would it be more convenient for the software to pack ADDRESSx and MASKx into a single register so they can be written with one store? (Maybe overoptimizing here because it will just be in init code)

Reply:
Do you mean packing them into, e.g., TARGET_IDx.ADDRESS and TARGET_IDx.MASK fields?

Reply:
Yes. so the code would be *TARGET_ID0 = (addr << ADDR_SHIFT) | (MASK << MASK_SHIFT);
But like I say probably over optimization.

Reply:
@mdhayter Would it be all right to pack all these into one register, TARGET_ID, with two ADDRESSx and MASKx fields?